### PR TITLE
Use C++17 attributes

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Currently, we are using non-standard attributes. With C++17, all the
attributes that we use regularly are now standardized. This commit
replaces the compiler-specific attributes with standard ones.

### CODE REVIEW

File: src/base/check.h [hunk 1]
```diff
@@ -54,14 +54,7 @@
 #define CVC5_PREDICT_TRUE(x) x
 #endif
 
-#ifdef __has_cpp_attribute
-#if __has_cpp_attribute(fallthrough)
 #define CVC5_FALLTHROUGH [[fallthrough]]
```
Comment (nafur) [inline src/base/check.h:57]: Why is this in `check.h`?

File: src/theory/theory.h [hunk 2]
```diff
@@ -440,18 +440,12 @@ class Theory : protected EnvObj
     EFFORT_LAST_CALL = 200
   }; /* enum Effort */
 
-  static inline bool standardEffortOrMore(Effort e) CVC5_CONST_FUNCTION
-  {
-    return e >= EFFORT_STANDARD;
-  }
-  static inline bool standardEffortOnly(Effort e) CVC5_CONST_FUNCTION
+  static bool standardEffortOrMore(Effort e) { return e >= EFFORT_STANDARD; }
+  static bool standardEffortOnly(Effort e)
```
Comment (nafur) [inline src/theory/theory.h:444]: Maybe this should be `constexpr` now?

File: src/base/check.h (hunk 1 above)
Comment (4tXJ7f) [reply src/base/check.h]: Yeah, I was wondering about that, too. Moved.

**Review by nafur — APPROVED** [review]
LGTM.

File: src/theory/theory.h (hunk 2 above)
Comment (4tXJ7f) [reply src/theory/theory.h]: Quick note: We have discussed this offline and came to the conclusion that `constexpr` does not apply here (and that the two methods `standardEffortOrMore()` and `standardEffortOnly()` are not really used).

**Pull request merged**